stellargraph/mapper/node_mappers.py

```diff
@@ -448,3 +454,132 @@ def flow_from_dataframe(self, node_targets):
+
+
+class FullBatchNodeSequence(Sequence):
+    """
+    Keras-compatible data generator to use with the Keras
+    methods :meth:`keras.Model.fit_generator`, :meth:`keras.Model.evaluate_generator`,
+    and :meth:`keras.Model.predict_generator`, for models that require full-batch training (e.g., GCN, GAT).
+
+    This class generated data samples for node inference models
+    and should be created using the `.flow(...)` method of
+    :class:`FullBatchNodeGenerator`.
+
+    These Generators are classes that capture the graph structure
+    and the feature vectors of each node.
+
+    """
+
+    def __init__(self, features, A, targets=None, sample_weight=None):
+        """
+
+        Args:
+            features: a matrix of node features of size (N x F), where N is the number of nodes in the graph, F is the node feature size
+            A: an adjacency matrix of the graph
+            targets: an optional array of node targets of size (N x C), where C is the target size (e.g., number of classes for one-hot class targets)
+            sample_weight: Optional Numpy array of weights for the node samples, used for weighting the loss function during training or evaluation.
+                You can either pass a flat (1D) Numpy array with the same length as the input features (1:1 mapping between weights and rows in features)
+        """
+        self.A = A
+        self.features = features
+        self.sample_weight = sample_weight
+        self.targets = targets
+
+    def __len__(self):
+        return 1
+
+    def __getitem__(self, batch_num):
+        return [self.features, self.A], self.targets, self.sample_weight
+
+
+class FullBatchNodeGenerator:
+    def __init__(self, G, name=None):
+        """
+        A data generator for node prediction with Homogeneous full-batch models, e.g., GCN, GAT
+
+        The supplied graph G should be a StellarGraph object that is ready for
+        machine learning. Currently the model requires node features for all
+        nodes in the graph.
+
+        Use the :meth:`flow` method supplying the nodes and (optionally) targets
+        to get an object that can be used as a Keras data generator.
+
+        Example::
+
+            G_generator = FullBatchNodeGenerator(G)
+            train_data_gen = G_generator.flow(node_ids, node_targets)
+
+        Args:
+            G (StellarGraphBase): a machine-learning StellarGraph-type graph
+            name (str): an optional name of the generator
+        """
+        if not isinstance(G, StellarGraphBase):
+            raise TypeError("Graph must be a StellarGraph object.")
+
+        self.graph = G
+        self.name = name
+
+        # Check if the graph has features
+        G.check_graph_for_ml()
+
+        # Create sparse adjacency matrix
+        self.node_list = list(G.nodes())
+        self.Aadj = nx.adjacency_matrix(G, nodelist=self.node_list)
+
+        # We need a schema to check compatibility with GraphSAGE, GAT, GCN
```

Comment:
Maybe remove reference to GraphSAGE? Can this generator be used with GraphSAGE?

Reply:
Fixed.

Reply:
fixed.